docs/appnotes/0009-storage-label-format.md

```diff
@@ -0,0 +1,83 @@
+
+In terms of requirements, we primarily want to enable the choice of get_urls based on properties such as bandwidth, latency, and cost.
+Unfortunately, many of these cannot be meaningfully signalled in a direct and universal way.
+If for no other reason, they are largely dependant on the intermediate infrastructure between the server and client.
+This will vary based on the physical and logical location of the client as much as the server.
+We must, therefore, provide enough information that the client can infer these properties themselves.
+This may be done by providing information about the location, and the type of storage.
+
+Unfortunately there is no one shared naming convention for resource locations across cloud providers.
+There are shared properties (regions, and availability zones), but these aren't universally available.
+Some cloud providers don't provide availability zones.
+And a local installation, or one behind a CDN might not even provide regions.
+So the signalling of these properties must be optional.
+Availability zones may also be semi-randomised in their naming to aid in evenly distributing load across cloud infrastructure.
+Zone naming is consistent within cloud accounts, though.
+Co-locating within a zone can be beneficial to performance.
+But this is only possible if a client knows it is within the same account as the server.
+It must, then, be possible for a client to either use or ignore availability zone based on whether it is within the same cloud account.
+This gives us provider, region, and availability zone as the important information that can be used to signal the location of storage.
+
+In addition to location, storage type is important for inferring the properties of the storage.
+Unfortunately, it is not possible or practical to directly compare similar types of products.
+For example, cloud providers may provide Object Storage.
+But they may provide multiple types of Object Storage.
+These may have vastly different properties.
+And there is often no universal naming conventions to describe these products in a way that is comparable between cloud providers.
+So while we need to signal this, we are not able to provide universal naming conventions.
+
+Finally, a more generic "store name" parameter is useful for human identification of stores, and for distinguishing stores which are otherwise identical.
+
+In summary.
+We must signal storage provider, region, availability zone, storage type, and store name.
+We cannot, unfortunately, signal any of these in consistently named and universally comparable ways.
+But we can specify a schema which allows a client to consistently decide which pieces of information are important to it.
+
+## Content
+
+Given all of the above, this Application Note recommends the following naming convention for the `get_urls` `label` parameter on flow segments:
+`<provider>.<region[optional]>.<availabilityZone[optional]>:<storeType>:<storeName>`
```

Comment:
Nit: This might flow better in the Markdown render as a code block, or pushed into a new paragraph

Reply:
Fixed in c20602a